packages/fast-components-react-base/src/foundation/foundation.ts

```diff
@@ -121,6 +121,43 @@ class Foundation<H, U, S> extends React.Component<H & U & IFoundationProps, S> {
         return componentClasses.concat(` ${this.props.className || ""}`).trim().replace(/(\s){2,}/g, " ") || null;
     }
 
+    /*
+     * Return an array of all nodes who's slot prop matches the provided slot.
+     * If no nodes are provided, `this.props.children` will be used
```

Comment:
should we also have the case that an assigned property may also contain children for examples `this.props.headerRegion`

Reply:
I don't follow

Reply:
discussed offline - this will accept any `React.ReactNode` - if `this.props.headerRegion` is a react node, then this will work